src/main/java/seedu/classcraft/parser/Parser.java

```diff
@@ -0,0 +1,200 @@
+
+public class Parser {
+    public String commandType;
+    public String userInputString;
+    public String userInstructions;
+    public StudyPlan studyPlan;
+
+    public Parser(String userInput) {
+        this.userInputString = userInput;
+        // this.studyPlan = studyPlan;
+        parseInstructions();
+    }
+
+    public Command parseInput() {
+        switch (commandType) {
+        case "help":
+            return new HelpCommand();
+        case "add":
+            String[] addModuleInfo = parseAdd();
+            // return new InvalidCommand();
+            return new AddCommand(addModuleInfo);
+        case "delete":
+            String deleteModuleCode = parseDelete();
+            // return new InvalidCommand();
+            return new DeleteCommand(deleteModuleCode);
+        case "confirm":
+            return new InvalidCommand();
+        //return new ConfirmCommand();
+        case "mc":
+            int semester = parseMC();
+            return new CalcCreditsCommand(semester - 1);
+        case "view":
+            String viewItems = parseView();
+            switch (viewItems) {
+            case "sample":
+                return new ViewSamplePlanCommand();
+            case "grad":
+                return new ViewGradReqCommand();
+            case "plan":
+                return new ViewCurrentPlanCommand();
+            // return new InvalidCommand();
+            default:
+                return new InvalidCommand();
+            }
+        case "exit":
+            return new ExitCommand();
+        case "spec":
+            String specItems = parseSpec();
+            return new SpecCommand(specItems);
+        default:
+            return new InvalidCommand();
+        }
+    }
+
+    //parses the user input into command type and instructions
+    //returns invalid command if command type is not found
+    public void parseInstructions() {
+        String[] instructions = userInputString.split(" ", 2);
+
+        if (!CommandList.isFound(instructions[0])) {
+            this.commandType = "invalid";
+            return;
+        }
+
+        try {
+            if (instructions.length == 1) {
+                if (!(instructions[0].equals("help") || instructions[0].equals("exit")
+                        || instructions[0].equals("confirm"))) {
+                    throw new IllegalArgumentException("OOPS!!! The description of a " +
+                            instructions[0] + " cannot be empty.");
+                }
+                this.commandType = instructions[0];
+            } else if (instructions.length == 2) {
+                if (instructions[1].isEmpty()) {
+                    throw new IllegalArgumentException("OOPS!!! The description of a " +
+                            instructions[0] + " cannot be empty.");
+                }
+                this.commandType = instructions[0];
+                this.userInstructions = instructions[1];
+            }
```

Comment:
see if you can improve SLAP here by extracting certain things out as separate methods